### PR TITLE
New inspection UI and plot editing

### PR DESCRIPTION
**Pull request by jeremyrsmith**
- Add a new inspection UI that comes up when you click certain places
- Update the plot editor to be able to do more stuff correctly
- Plot editor can now save plots to a Vega cell

The "Value" column of the symbol table is gone, and now clicking a row in the symbol table brings up the inspection UI. It gives you the different available views of a variable, including a nicer plot editor and data viewer (not much was done to the data viewer).

It still needs some work, and some refactoring has to be done here because it's getting rather spaghetti-ish. I'll sit down and work out how to restructure the front-end stuff soon. But I wanted to get this feature out because it is pretty useful.

One thing that this brings to light is that we need a way to hide the code of code cells. A vega spec is a giant wall of text that becomes useless noise when you're not actively editing it.

### CODE REVIEW

File: polynote-frontend/polynote/plot_editor.js [hunk 1]
```diff
@@ -121,15 +153,16 @@ export class PlotEditor extends EventTarget {
         });
 
         this.xAxisDrop.addEventListener('dragenter', evt => {
-           if (this.draggingEl.classList.contains('dimension')) {
+           if (this.draggingEl.classList.contains(this.correctXType)) {
                this.xAxisDrop.classList.add('drop-ok');
            } else {
+               console.log(this.draggingEl);
```
Comment (indig) [inline polynote-frontend/polynote/plot_editor.js]: Is this left over from debugging?

File: polynote-frontend/polynote/plot_editor.js [hunk 2]
```diff
@@ -179,17 +222,50 @@ export class PlotEditor extends EventTarget {
         // TODO - show a UI to let you
```
Comment (indig) [inline polynote-frontend/polynote/plot_editor.js]: not part of the PR but TODO can be removed.

File: polynote-frontend/polynote/plot_editor.js [hunk 3]
```diff
@@ -223,28 +303,35 @@ export class PlotEditor extends EventTarget {
...
         this.xDimension = field;
+        this.xAxisDrop.classList.add('nonempty');
         const label = this.xAxisDrop.querySelector('.label');
-        label.innerHTML = '';
-        label.appendChild(document.createTextNode(field.name));
+        [...label.querySelectorAll('.numeric, .dimension')].forEach(node => node.parentNode.removeChild(node));
+        label.appendChild(span([this.correctXType], [field.name]));
+    }
+
+    addXField(from) {
```
Comment (indig) [inline polynote-frontend/polynote/plot_editor.js]: What's the idea behind both `addXField` and `set xField`? (also, is `this.xField` used anywhere?)

File: polynote-frontend/polynote/plot_editor.js (hunk 1 above)
Comment (jeremyrsmith) [reply polynote-frontend/polynote/plot_editor.js]: Yup, fixed

File: polynote-frontend/polynote/plot_editor.js (hunk 3 above)
Comment (jeremyrsmith) [reply polynote-frontend/polynote/plot_editor.js]: Yeh, kind of a mess... I removed the setter thing and just changed it to `setXField`, and removed `addXField`.

**Comment by indig** [comment]
I believe this resolves https://github.com/polynote/polynote/issues/239 and https://github.com/polynote/polynote/issues/238 ?
